runtime-modules/content/src/lib.rs

```diff
@@ -3589,6 +3599,33 @@ impl<T: Config> Module<T> {
             ChannelOwner::CuratorGroup(..) => Ok(ChannelFundsDestination::CouncilBudget),
         }
     }
+    //Weight functions
+
+    // Calculates weight for channel_creation_weight extrinsic.
+    fn channel_creation_weight(params: &ChannelCreationParameters<T>) -> Weight {
+        // new_meta.
+        let a = params.meta.as_ref().map_or(0, |v| v.len()) as u32;
+
+        //collaborators
+        let b = params
+            .collaborators
+            .iter()
+            .fold(0, |acc, (_, permissions)| acc + permissions.len() + 1) as u32;
```

Comment:
In the benchmark code `b` represents the number of collaborators, but here you're passing the number of collaborators + the number of permissions that each collaborator has

Reply:
I'm trying to represent the worst case scenario.
Collaborators are composed by a collection inside a collection
`pub collaborators: BTreeMap<MemberId, ChannelAgentPermissions>`
iterating that will make a node to process the ChannelAgentPermissions for each collaborator.
Am I mistaken?

Reply:
Collaborator is just one entry in the map, regardless of `ChannelAgentPermissions` size.
Your benchmarks already use fixed `CHANNEL_AGENT_PERMISSIONS` which should represent a worst-case-scenario per collaborator. So `b` is the number of entries in `collaborators` map (`collaborators.len()`) and for each b, the worst case scenario for the size of `ChannelAgentPermissions` is assumed by the benchmarks.